slow_odgi/to_json.py

```diff
@@ -0,0 +1,106 @@
+import sys
+import json
+from json import JSONEncoder
+import mygfa
+import preprocess
+
+
+class SegmentEncoder(JSONEncoder):
+    def default(self, o):
+        return o.__dict__
```

Comment:
Maybe `o.asdict()` would be clearer? It's documented for dataclasses, whereas `__dict__` is more of a subtle Python-internals thing:
https://docs.python.org/3/library/dataclasses.html#dataclasses.asdict